### PR TITLE
[Beam-1280] remove label arguments from ptransforms

### PR DESCRIPTION
**Pull request by sb2nov**
R: @aaltay PTAL

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by sb2nov** [comment]
retest this please

File: sdks/python/apache_beam/transforms/combiners.py [hunk 1]
```diff
@@ -476,7 +476,7 @@ def __init__(self, label='ToList'):
     super(ToList, self).__init__(label)
 
   def expand(self, pcoll):
-    return pcoll | core.CombineGlobally(self.label, ToListCombineFn())
+    return pcoll | self.label >> core.CombineGlobally(ToListCombineFn())
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/combiners.py]: I think we should remove these `self.label` things. (Not necessarily in this PR, but definitely a backward incompatible change.)

How would this work in for example
`label1 >> ToList(label2)`

File: sdks/python/apache_beam/transforms/ptransform.py [hunk 2]
```diff
@@ -639,15 +633,9 @@ def display_data(self):
...
-      label, self._args = None, args[1:]
-    elif args and isinstance(args[0], str):
-      label, self._args = args[0], args[1:]
-    else:
-      label, self._args = None, args
+    self._args = args
     self._kwargs = kwargs
-    # We know the label now, so initialize the super-class.
-    super(CallablePTransform, self).__init__(label=label)
+    super(CallablePTransform, self).__init__()
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/ptransform.py]: `super` call can be the first thing now.

File: sdks/python/apache_beam/transforms/combiners.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/transforms/combiners.py]: yeah I think we should remove label from __init__ of any derived class. I'll address that in another PR.

Comment (sb2nov) [inline sdks/python/apache_beam/transforms/combiners.py]: To answer that question I think label1 will override as that is applied after the label2 computation is already done.

File: sdks/python/apache_beam/transforms/core.py [hunk 3]
```diff
@@ -592,8 +590,8 @@ class ParDo(PTransformWithSideInputs):
   the argument lists.
   """
 
-  def __init__(self, fn_or_label, *args, **kwargs):
-    super(ParDo, self).__init__(fn_or_label, *args, **kwargs)
+  def __init__(self, fn, *args, **kwargs):
```
Comment (sb2nov) [inline sdks/python/apache_beam/transforms/core.py]: Done.

File: sdks/python/apache_beam/transforms/ptransform.py (hunk 2 above)
Comment (sb2nov) [inline sdks/python/apache_beam/transforms/ptransform.py]: Done

**Comment by sb2nov** [comment]
I think the PR is long enough now that I'll make more label changes in another from now to keep the review process sane.

**Comment by aaltay** [comment]
LGTM. Thank you @sb2nov

**Pull request closed without merging**